zomes/holo_acorn/code/src/lib.rs

```diff
@@ -220,7 +220,17 @@ mod holo_acorn {
                             Err("only the same agent can create your profile".into())
                         }else {Ok(())}
                     },
-                    _=> Ok(())
+                    hdk::EntryValidationData::Modify{
+                        old_entry,..}=>{
+                        if  old_entry.address!=AGENT_ADDRESS.to_string(){
+                            Err("only the same agent can modify your profile".into())
+                        }else {Ok(())}
```

Comment:
we should also make sure that the new entry uses the same `new_entry.address`... that it also equals AGENT_ADDRESS